### PR TITLE
Feat: more configuration for format command

### PR DESCRIPTION
**Pull request by z3z1ma**
Not enough configurability. Expose very nice implementation in Generator to the benefit of all users while preserving existing behavior by default.

### CODE REVIEW

**Comment by eakmanrq** [comment]
@z3z1ma Can you also add these to the magic command?

https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/magics.py#L627

**Comment by z3z1ma** [comment]
You bet! Done @eakmanrq

File: sqlmesh/magics.py [hunk 1]
```diff
@@ -618,16 +618,49 @@ def rewrite(self, context: Context, line: str, sql: str) -> None:
...
+        help="The max number of characters in a segment before creating new lines in pretty mode.",
+        default=80,
+    )
     @line_magic
     @pass_sqlmesh_context
     def format(self, context: Context, line: str) -> None:
         """Format all SQL models."""
         args = parse_argstring(self.format, line)
-        context.format(args.transpile, args.new_line)
+        context.format(**vars(args))
```
Comment (tobymao) [inline sqlmesh/magics.py:663]: what's vars?

Comment (z3z1ma) [reply sqlmesh/magics.py:663]: The arg parser returns a Namespace. `vars` accesses the `__dict__` attribute basically. On namespaces, the is well defined behavior to convert it to a dict.

Comment (z3z1ma) [reply sqlmesh/magics.py:663]: See stdlib docs mention the same

https://docs.python.org/3/library/argparse.html#the-namespace-object

**Pull request merged**